### PR TITLE
Fix #8856: Keep transparent flag

### LINKED ISSUE #8856: Keep `transparent` modifier
`transparent` modifier must be kept after desugaring. Currently, it is impossible to know if a definition is transparent or not after it is typed. 

This is a blocker for #8850 and  #7825. We should also have it in TASTy for tools to access this information.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: tasty/src/dotty/tools/tasty/TastyFormat.scala [hunk 1]
```diff
@@ -248,7 +248,7 @@ Standard Section: "Comments" Comment*
 object TastyFormat {
 
   final val header: Array[Int] = Array(0x5C, 0xA1, 0xAB, 0x1F)
-  val MajorVersion: Int = 22
```
Comment (odersky) [inline tasty/src/dotty/tools/tasty/TastyFormat.scala:251]: Don't we need to pickle/unpickle TRANSPARENT?

**Comment by nicolasstucki** [comment]
We cannot detect if a `def` is marked with `transparent` by looking it's RHS because it is ambiguous. Furthermore, we can't use that strategy to detect if a `val` was marked as transparent.

**Comment by nicolasstucki** [comment]
The alternative would be to add an annotation.

File: tasty/src/dotty/tools/tasty/TastyFormat.scala (hunk 1 above)
Comment (nicolasstucki) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala:251]: Done

**Comment by odersky** [comment]
What is a `transparent` val?

**Comment by odersky** [comment]
I am beginning to think that the whole introduction of `transparent` was a mistake. Maybe we should have stuck with the special type syntax? The problem with modifiers is feature creep - they tend to go everywhere.

**Comment by nicolasstucki** [comment]
Still, we need a way for tools to know if an inline is transparent, regardless of the encoding.

**Comment by odersky** [comment]
Yes, but an inline could be transparent if its RHS is not a type ascription. The question is: can we always see that?

**Pull request closed without merging**